karabo_data/components.py

```diff
@@ -78,17 +82,35 @@ class MPxDetectorBase:
     """
 
     _source_re = re.compile(r'(?P<detname>.+)/DET/(\d+)CH')
+    module_shape = (0, 0)  # Override in subclass
 
-    def __init__(self, data: DataCollection, detector_name=None, modules=None):
+    def __init__(self, data: DataCollection, detector_name=None, modules=None,
+                 *, min_modules=1):
         if detector_name is None:
             detector_name = self._find_detector_name(data)
 
         source_to_modno = self._identify_sources(data, detector_name, modules)
 
-        self.data = data.select([(src, '*') for src in source_to_modno])
+        data = data.select([(src, '*') for src in source_to_modno])
         self.detector_name = detector_name
         self.source_to_modno = source_to_modno
 
+        # pandas' missing-data handling converts the data to floats if there
+        # are any gaps - so fill them with 0s and convert back to uint64.
+        mod_data_counts = pd.DataFrame({
+            src: data.get_data_counts(src, 'image.data')
+            for src in source_to_modno
+        }).fillna(0).astype(np.uint64)
+
+        # Sanity check: all non-zero counts should be equal
+        data_count_values = set(mod_data_counts.values.flatten()) - {0}
+        if len(data_count_values) > 1:
+            raise ValueError("Inconsistent frame counts: {}"
+                             .format(data_count_values))
+
+        self.frames_per_train = data_count_values.pop()
```

Comment:
I think you might want to try only if data_count_values is not empty. If not it will through an error

Reply:
Thanks, good point.